PICMI_Python/lasers.py

```diff
@@ -65,6 +66,65 @@ def __init__(self, wavelength, waist, duration,
         self.handle_init(kw)
 
 
+class PICMI_AnalyticLaser(_ClassWithInit):
+    """
+    Specifies a laser with an analytically described distribution
+      - field_expression: Analytic expression describing the electric field of the laser(string) [V/m]
+                            Expression should be in terms of the position, 'X', 'Y', in the plane orthogonal
+                            to the propagation direction, and 't' the time. The expression should describe
+                            the full field, including the oscillitory component.
+                            Parameters can be used in the expression with the values given as keyword arguments.
+      - propagation_direction=[0,0,1]: Direction of propagation (unit vector) [1]
+      - polarization_direction=[1,0,0]: Direction of polarization (unit vector) [1]
+
+      Even though the parameters below should be built into the expression, some codes require
+      specified values for numerical purposes.
+
+      - wavelength: Laser wavelength
+      - amax: Maximum normalized vector potential
+              Specify either amax or Emax (Emax takes precedence).
+      - Emax: Maximum amplitude of the laser field [V/m]
+              Specify either amax or Emax (Emax takes precedence).
+    """
+    def __init__(self, field_expression,
+                 wavelength,
+                 propagation_direction = [0., 0., 1.],
+                 polarization_direction = [1., 0., 0.],
+                 amax = None, 
+                 Emax = None,
+                 **kw):
+
+        k0 = 2.*math.pi/wavelength
+        if Emax is None:
+            from scipy import constants
+            Emax = amax*constants.electron_mass*constants.speed_of_light**2*k0/constants.elementary_charge
+        if amax is None:
+            from scipy import constants
+            amax = Emax/(constants.electron_mass*constants.speed_of_light**2*k0/constants.elementary_charge)
```

Comment:
I was wondering if it would help users, if we add an error message if neither `Emax` nor `amax` are defined.
```suggestion
        if (Emax is None) and (amax is None):
                raise Exception('Emax and/or amax need to be defined (Emax has precedence)')
        elif Emax is None:
            from scipy import constants
            Emax = amax*constants.electron_mass*constants.speed_of_light**2*k0/constants.elementary_charge
        elif amax is None:
            from scipy import constants
            amax = Emax/(constants.electron_mass*constants.speed_of_light**2*k0/constants.elementary_charge)
```

Reply:
Yes, this is a good idea. Though elsewhere, these kinds of checks are done with assert, i.e.
assert Emax is not None or amax is not None, "One of Emax or amax must be specified"

Reply:
Ah yes yes, I looked for error or abort. I forgot to check if we could use assert in the python files. Please feel free to make that change and I will learn from it for any future changes I make. Thank you @dpgrote !

Reply:
I've updated adding the asserts (adding one for Gaussian as well).